pkg/didcomm/common/service/action.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	"sync"
+)
+
+// Action thread-safe action register structure
+type Action struct {
+	mu    sync.RWMutex
+	event chan<- DIDCommAction
+}
+
+// GetActionEvent returns event action channel
+func (a *Action) GetActionEvent() chan<- DIDCommAction {
+	a.mu.RLock()
+	e := a.event
+	a.mu.RUnlock()
+	return e
+}
+
+// RegisterActionEvent on protocol messages.
+// The consumer need to invoke the callback to resume processing.
+func (a *Action) RegisterActionEvent(ch chan<- DIDCommAction) error {
```

Comment:
Can we retain [this](https://github.com/hyperledger/aries-framework-go/pull/449/files#diff-f761c91e753eba1c42a68076df06ce12L167) existing GoDoc line from original method ?